### DISCUSSION TITLE
GA4's data is being blocked from my Hub's site

### DISCUSSION THREAD

**Discussion by VeroBleu** [mozilla/hubs#5520] [Hubs Cloud Questions] [answered]
Hi there! I'm having trouble with Google Tag Manager & Analytics, Hub's code is kinda blocked and no data is being received from the website to our Google Analytics platform.

![Bug Google Tag Manager](https://user-images.githubusercontent.com/107501637/173640173-443cc88a-3e9c-4f6f-9fc3-74937f600c38.jpeg)

The website: https://sopart.click/764M4qU/sopart-2021

Can you help me with this? TY!

**Comment by matthewbcool** [comment, marked as answer]
You can make edits to the CSP in Server Settings through the Admin Panel --> Server Settings --> Advanced Tab.

Some have reported needing to re-deploy in order to get results. 
See also: https://github.com/mozilla/hubs/discussions/5455

**Reply by VeroBleu** [reply]
This was very helpful. TYVM!

**Reply by VeroBleu** [reply]
@matthewbcool hello again! I'm still having issues as GA4 still not receiving data from the hub's site.
The code inserted was:

<script async src="https://www.googletagmanager.com/gtag/js?id=G-2J0W4QB5E9"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-2J0W4QB5E9');
</script>

**But still, GA4's not receiving visitors' data, do you know what may be going on? Also checked the #5455 but it didn't resolve my issue.** Thank you in advance!

**Reply by matthewbcool** [reply]
@VeroBleu could you let us know what you are writing in the CSP settings (Admin Panel --> Server Settings --> Advanced Tab) and where you are injecting this script? Thank you!

**Reply by VeroBleu** [reply]
TY @matthewbcool, here's the full SS:

![sopart click_admin](https://user-images.githubusercontent.com/107501637/185202103-9d77976c-596b-42c3-8230-d5ad1f708cd7.png)

**Comment by thomasLopvetVR** [comment]
Update from July 2023 : the answer is not working and the the standard GA ID is not working anymore.
I tried everything but even adding 'unsafe-inline' in the CSP is not working.
